app/views/onboardings/_newsletter.html.erb

```diff
@@ -0,0 +1,63 @@
+<% if feature_flag_enabled?(:onboarding_newsletter_content) &&
+      Settings::General.onboarding_newsletter_content_processed_html.present? &&
+      Settings::General.onboarding_newsletter_opt_in_head.present? &&
+      Settings::General.onboarding_newsletter_opt_in_subhead.present? %>
+
+  <%= Settings::General.onboarding_newsletter_content_processed_html.html_safe %>
+
+  <form>
+    <fieldset>
+      <div class="flex email_newsletter mt-5 pt-4 ">
+        <div class="flex h-5 ml-3">
+          <input type="checkbox" id="email_newsletter" name="email_newsletter" class="w-4 h-4 mt-1 crayons-checkbox">
+        </div>
+        <div class="ml-3">
+          <label for="email_newsletter" class="font-medium text-gray-900 dark:text-gray-300">
+            <p class="onboarding_newsletter_opt_in_head" style="font-weight: 600"><%= Settings::General.onboarding_newsletter_opt_in_head %></p>
+            <p class="onboarding_newsletter_opt_in_subhead mt-1 pb-4"><%= Settings::General.onboarding_newsletter_opt_in_subhead %></p>
+          </label>
+        </div>
+      </div>
+    </fieldset>
+  </form>
+<% elsif feature_flag_enabled?(:onboarding_newsletter_content) %>
```

Comment:
I like the way these cases differentiate between “behavior as before,” “feature flag with customized newsletter”, and “feature flag without newsletter customization”. 

Am I reading correctly that it defaults to this elsif when the feature is on and some, but not all newsletter settings are customized?

Reply:
> Am I reading correctly that it defaults to this elsif when the feature is on and some, but not all newsletter settings are customized?

@dukegreene I'm pretty sure we're on the same page, but to be explicit here... there are three modes for this partial:

* Flag off — original behavior
* Flag on, not fully configured — new "unconfigured" or "default" mode
* Flag on, fully configured — use the configuration settings

(The order they appear in the partial is inverted from this listing.)

I'm maybe being a bit cautious, but my idea was partly to constrain the parameters we're trying to design around (what if there's no heading? what if there's no subheading? etc...) and partly to prevent releasing anything that's not quite "ready for prime time". I _think_ we've managed to wrangle a design that [degrades pretty well](https://github.com/forem/forem/issues/19987#issuecomment-1729135589), but I'm still a little cautious about letting anything "peek out" before we're ready.

Also — and I _know_ these are "famous last words", but — I don't expect it to stay in this form for very long.

Reply:
Ahhh! Thanks for this thorough explanation. I ran into the same confusion and now understand why. Reasoning makes sense for initial rollout!

Reply:
Maybe smth like would be crearer (but more nesting). I was a bit confused while reviewing too.

```ruby
<% if feature_flag_enabled?(:onboarding_newsletter_content) %>
  <% if Settings::General.onboarding_settings %>
     use the configuration settings
  <% else %>
     new "unconfigured" or "default" mode
  <% end %>
<% else %>
   original behavior
<% end %>
```